src/profile.jl

```diff
@@ -0,0 +1,321 @@
+struct FeProfile{T<:AbstractFloat}  # derived model with the j'th fixed-effects coefficient held constant
+    m::LinearMixedModel{T}          # copy of original model after removing the j'th column from X
+    y₀::Vector{T}                   # original response vector
+    xⱼ::Vector{T}                   # the column that was removed from X
+    j::Integer
+end
+
+"""
+    Base.copy(ReMat{T,S})
+
+Return a shallow copy of ReMat.
+
+A shallow copy shares as much internal storage as possible with the original ReMat.
+Only the vector `λ` and the `scratch` matrix are copied.
+"""
+function Base.copy(ret::ReMat{T,S}) where {T,S}
+    return ReMat{T,S}(ret.trm,
+        ret.refs,
+        ret.levels,
+        ret.cnames,
+        ret.z,
+        ret.wtz,
+        copy(ret.λ),
+        ret.inds,
+        ret.adjA,
+        copy(ret.scratch))
+end
+
+## FIXME: also create a shallow copy of a LinearMixedModel object that performs a shallow copy of the reterms and the optsum.
+##   Probably don't bother to copy the components of L as we will always assume that an updateL! call precedes a call to
+##   objective.
+
+"""
+    _copy_away_from_lowerbd!(sink, source, bd; incr=0.01)
+
+Replace `sink[i]` by `max(source[i], bd[i] + incr)`.  When `bd[i] == -Inf` this simply copies `source[i]`.
+"""
+function _copy_away_from_lowerbd!(sink, source, bd; incr=0.01)
+    for i in eachindex(sink, source, bd)
+        @inbounds sink[i] = max(source[i], bd[i] + incr)
+    end
+    return sink
+end
+
+function σvals(m::LinearMixedModel{T}) where {T}
+    @compat (; σ, reterms) = m
+    isone(length(reterms)) && return σvals(only(reterms), σ)
+    return (collect(Iterators.flatten(σvals.(reterms, σ)))...,)
+end
+
+function FeProfile(m::LinearMixedModel, j::Integer)
+    Xy = m.Xymat.xy
+    xcols = collect(axes(Xy, 2))
+    ycol = pop!(xcols)
+    notj = deleteat!(xcols, j) # indirectly check that j ∈ xcols
+    y₀ = Xy[:, ycol]
+    xⱼ = Xy[:, j]
+    feterm = FeTerm(Xy[:, notj], m.feterm.cnames[notj])
```

Comment:
would it be possible to use a view here?

Reply:
I think I was copying those vectors and matrices out of an abundance of caution, because I didn't want to accidentally modify values in the original model structure.

Would it be okay to "stick a pin" in this and remind ourselves to follow up later on whether views can be used instead?

That reminds me that we may want to consider setting some of the fields in the larger structures as `const` (I think the change allowing that was in Julia v1.8.0 so we may need to put that inside a check on version numbers.)